Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BinaryVsTextPerfTest.cs

```
@@ -0,0 +1,342 @@
+
+            while (queue.Count > 0)
+            {
+                ITrace node = queue.Dequeue();
+                if (node.Data.ContainsKey("Query Metrics"))
+                {
+                    return node;
+                }
+                foreach (ITrace child in node.Children)
+                    queue.Enqueue(child);
+            }
+            return null;
+        }
+
+        private ITrace FindQueryTransitMetrics(ITrace trace)
+        {
+            Queue<ITrace> queue = new Queue<ITrace>();
+            queue.Enqueue(trace);
+
+            while (queue.Count > 0)
+            {
+                ITrace node = queue.Dequeue();
+                if (node.Data.ContainsKey("Client Side Request Stats") & node.Name == "Microsoft.Azure.Documents.ServerStoreModel Transport Request")
+                {
+                    return node;
+                }
+                foreach (ITrace child in node.Children)
+                    queue.Enqueue(child);
+            }
+            return null;
+        }
+
+        private ITrace FindQueryClientMetrics(ITrace trace, string name)
+        {
+            Queue<ITrace> queue = new Queue<ITrace>();
+            queue.Enqueue(trace);
+
+            while (queue.Count > 0)
+            {
+                ITrace node = queue.Dequeue();
+                if (node.Name == name)
+                {
+                    return node;
+                }
+                foreach (ITrace child in node.Children)
+                    queue.Enqueue(child);
+            }
+            return null;
+        }
+
+        private double CalculateAverage(List<double> AvgList)
+        {
+            AvgList = AvgList.GetRange(this.WarmupRounds, AvgList.Count - this.WarmupRounds);
+            return Queryable.Average(AvgList.AsQueryable());
+        }
+
+        private double CalculateMedian(List<double> MedianList)
+        {
+            MedianList = MedianList.GetRange(this.WarmupRounds, MedianList.Count - this.WarmupRounds);
```

> MedianList = MedianList.GetRange(this.WarmupRounds, MedianList.Count - this.WarmupRounds); [](http://example.com/codeflow?start=11&length=91)

same as above:

Avoid mixing concerns: a method that calculates median should not silently ignore part of its input, and it should not mutate its input